sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/kinesis/KinesisReadSchemaTransformProvider.java

```diff
@@ -0,0 +1,223 @@
+
+public class KinesisReadSchemaTransformProvider
+    extends TypedSchemaTransformProvider<
+        KinesisReadSchemaTransformProvider.KinesisReadSchemaTransformConfiguration> {
+  public static final ImmutableSet<String> SUPPORTED_INITIAL_POSITIONS =
+      ImmutableSet.of("LATEST", "TRIM_HORIZON", "AT_TIMESTAMP");
+
+  public static final ImmutableSet<String> SUPPORTED_FORMATS = ImmutableSet.of("AVRO", "JSON");
+
+  public static final TupleTag<Row> MAIN_TAG = new TupleTag<Row>("main");
+  public static final TupleTag<Row> ERROR_TAG = new TupleTag<Row>("errors");
+  public static final Schema ERROR_SCHEMA =
+      Schema.builder().addStringField("error").addNullableByteArrayField("row").build();
+
+  @Override
+  public @UnknownKeyFor @NonNull @Initialized String identifier() {
+    return "beam:schematransform:org.apache.beam:kinesis_read:v1";
+  }
+
+  @Override
+  public @UnknownKeyFor @NonNull @Initialized List<@UnknownKeyFor @NonNull @Initialized String>
+      inputCollectionNames() {
+    return Collections.emptyList();
+  }
+
+  @Override
+  public @UnknownKeyFor @NonNull @Initialized List<@UnknownKeyFor @NonNull @Initialized String>
+      outputCollectionNames() {
+    return Lists.newArrayList("output", "errors");
+  }
+
+  @Override
+  protected @UnknownKeyFor @NonNull @Initialized Class<KinesisReadSchemaTransformConfiguration>
+      configurationClass() {
+    return KinesisReadSchemaTransformConfiguration.class;
+  }
+
+  @Override
+  protected @UnknownKeyFor @NonNull @Initialized SchemaTransform from(
+      KinesisReadSchemaTransformConfiguration configuration) {
+    if (!SUPPORTED_INITIAL_POSITIONS.contains(configuration.getInitialPositionInStream())) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Unknown initial position in stream %s. Supported values: %s",
+              configuration.getInitialPositionInStream(), SUPPORTED_INITIAL_POSITIONS));
+    }
+    if (!SUPPORTED_FORMATS.contains(configuration.getFormat().toUpperCase())) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Unknown data format %s. Supported values: %s",
+              configuration.getFormat(), SUPPORTED_FORMATS));
+    }
+    return new KinesisReadSchemaTransform(configuration);
+  }
+
+  @AutoValue
+  @DefaultSchema(AutoValueSchema.class)
+  public abstract static class KinesisReadSchemaTransformConfiguration implements Serializable {
+    public abstract String getFormat();
+
+    public abstract String getSchema();
+
+    public abstract String getStreamName();
+
+    @SchemaFieldDescription("Supported values are [\"LATEST\", \"TRIM_HORIZON\", \"AT_TIMESTAMP\"]")
+    public abstract String getInitialPositionInStream();
+
+    @SchemaFieldDescription(
+        "Access key ID for AWS credentials. See https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html")
+    public abstract String getCredentialsAccessKeyId();
```

Comment:
This should be nullable to allow using the default credentials provider chain. Also same below...

Reply:
Alternatively this could be implemented similarly to `AwsOptions#getAwsCredentialsProvider()` using the serialization mechanism used there.